### PR TITLE
[instance] move definition of all OT core objects in `Instance`

### PR DESCRIPTION
**Pull request by abtink**
This commit moves all the OT core member variable definitions into
`Instance` class (from `ThreadNetif`).

### CODE REVIEW

**Comment by bukepo** [comment]
May I know the motivation of moving objects from `ThreadNetif` into `otInstance`?

**Comment by abtink** [comment]
> May I know the motivation of moving objects from ThreadNetif into otInstance?

- Simplify the code. 
- Help with inlining/optimizing done by toolchain which can help reduce code size.

File: src/core/net/ip6.cpp [hunk 1]
```diff
@@ -1293,7 +1293,7 @@ Error Ip6::HandleDatagram(Message &aMessage, Netif *aNetif, const void *aLinkMes
 #endif
 
         // `SendMessage()` takes custody of message in the success case
-        SuccessOrExit(error = Get<ThreadNetif>().SendMessage(aMessage));
+        SuccessOrExit(error = Get<MeshForwarder>().SendMessage(aMessage));
```
Comment (simonlingoogle) [inline src/core/net/ip6.cpp:1296]: I think we can still use ThreadNetif for sending messages?

It's natural to say that messages are sent via the Thread network interface. 

Thoughts?

**Review by simonlingoogle — APPROVED** [review]
LGTM 

It makes perfect sense.

File: src/core/net/ip6.cpp (hunk 1 above)
Comment (abtink) [reply src/core/net/ip6.cpp]: I re-added it back in latest push. I agree it makes sense to want to send message via Thread netif  (at some point, we may combine `Netif and `ThreadNetif` into one class and we can re-consider this). Also it did not fit well with the main change in this PR.

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**